assets/js/src/index.js

```diff
@@ -1,6 +1,24 @@
-// Initialize tracking for classic WooCommerce pages
-import { trackClassicPages } from './integrations/classic';
-window.wcgai.trackClassicPages = trackClassicPages;
+import { createTracker } from './tracker';
+import { classicTracking } from './integrations/classic';
+import { blocksTracking } from './integrations/blocks';
 
-// Initialize tracking for Block based WooCommerce pages
-import './integrations/blocks';
+// Wait for DOMContentLoaded to make sure event data is in place.
+if ( document.readyState === 'loading' ) {
+	document.addEventListener(
+		'DOMContentLoaded',
+		eventuallyInitializeTracking
+	);
+} else {
+	eventuallyInitializeTracking();
+}
+function eventuallyInitializeTracking() {
+	if ( ! window.ga4w ) {
+		throw new Error(
+			'Google Analytics for WooCommerce: Configuration and tracking data not found.'
+		);
+	}
+	const eventHandler = createTracker( window.ga4w.settings );
```

Comment:
>Just a minor point of confusion, from my impression all that createTracker does is setup an eventHandler, does createEventHandler make more sense?

I have hard time with thinking of the names ;) as we have
1. [the `gtag` (or renamed) function](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/398/files#diff-efaa51133b4f4fcb6d50a840b6c0e00dd53244078149890409ddbe43ce3d7408R32)  - "tracking function"?
2. [Function to be fed with event data](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/398/files#diff-efaa51133b4f4fcb6d50a840b6c0e00dd53244078149890409ddbe43ce3d7408R29), that formats it and calls 1. This function is given to Block action - the actual "event handler"?
3. [Function that takes event name](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/398/files#diff-efaa51133b4f4fcb6d50a840b6c0e00dd53244078149890409ddbe43ce3d7408R22), picks the formatted for it and produces a function 2. - "create/get event handler"?
4. [Function that makes the global configuration](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/398/files#diff-efaa51133b4f4fcb6d50a840b6c0e00dd53244078149890409ddbe43ce3d7408R11) to produce 3.
 - "create event handler creator"?
 
 So I think of a two sets of new names 
 1. "tracker" plus its `trackerFunctionName`
 2. `trackerEventHandler`
 3. `eventHandler`
 4. `createEventHandler`
 
 or 

 1. "tracker"
 2. `eventHandler`
 3. `getEventHandler`
 4. `createEventHandlers` - but technically, it does not create event handlers, it creates a function to create them.

or we could change it even more, as instead of creating a new function every time the function 3. is called. we could create all of them as once 4 is called and put it in a map, something like:

```js
const eventHandlers = careateEventHandlers( window.ga4w.settings ); // new EventHandlers(  window.ga4w.settings  );
classicTracking( eventHandler, window.ga4w.data );
blocksTracking( eventHandler );
// ...
addUniqueAction(`foo`, 'bar', eventHandlers.get( 'view_item' ) ); 
// or
addUniqueAction(`foo`, 'bar', eventHandlers[ 'view_item' ] );
// ...
eventHandlers.get( 'add_to_cart' )( { product: productToHandle } );
eventHandlers[ 'add_to_cart' ]( { product: productToHandle } );
```
Which is basically turning 3. to a more native, map-accessor function:
1. "tracker"
 2. `eventHandler`
 3. `Map.prototype.get` / `Object`'s property accessor
 4. `createEventHandlers`

Reply:
> But it doesn't actually use them till an event is handled (which can be much later). I guess in this case that's not useful to change because the `classicTracking` does rely on the data being available on construction.

I'm not sure if I get your point here.

My point was to pass the config once in the entry. Then, all the code that binds the event handlers to actions or DOM events would not have to keep passing it. Would not have to access it from the global scope or cascade it through all the parent functions.

Reply:
> I'm not sure if I get your point here.

Sorry about the confusion there. I was analyzing the `createTracker` first and realized that it didn't need to throw the error `Google Analytics for WooCommerce: Configuration and tracking data not found.`, as it only needs the settings later. But then I inspected classicTracker and realized that there the data is needed right away. So in theory we could load createHandler without throwing the warning but it's not much use to do so as it's only used by the classicTracking and blockTracking. So to summarize the comment was my train of thought but not valid to apply.

Reply:
> I have hard time with thinking of the names ;)

Naming is definitely a hard topic, but it was just a minor point, no use in doing that much refactoring for my sake.

Reply:
Speaking of function names, WDYT of https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/398/commits/cdb60c08a9cb62f351f5d324472ad62b6eef7105 ?

Reply:
I think that's a bit clearer. I don't see any need to refactor beyond that, as it will just be nitpicking.